### PR TITLE
Op 278 - Add sylius 1.13 support

### PR DESCRIPTION
**Pull request by SzymonKostrubiec**
+ Apply changes from community pull request
+ Fixed a.gitignore  file
+ Added builds for nod 18.x
+ Restore google chrome driver in behat tests

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,85 +1,91 @@
...
-    "type": "sylius-plugin",
-    "keywords": [
-        "sylius",
-        "sylius-plugin",
-        "sylius-cms"
-    ],
-    "description": "CMS plugin for Sylius applications.",
-    "license": "MIT",
-    "require": {
-        "php": "^8.0",
```
Comment (marekrzytki) [inline composer.json:12]: Why PHP requirement is removed?

File: composer.json [hunk 2]
```diff
@@ -1,85 +1,91 @@
...
+  "type": "sylius-plugin",
+  "keywords": [
+    "sylius",
+    "sylius-plugin",
+    "sylius-cms"
+  ],
+  "description": "CMS plugin for Sylius applications.",
+  "license": "MIT",
+  "require": {
+    "sylius/sylius": "^1.12 || ^1.13",
```
Comment (marekrzytki) [inline composer.json:12]: "^1.12.0 || ^1.13.0",

File: phpunit.xml.dist [hunk 3]
```diff
@@ -15,7 +15,5 @@
         <server name="KERNEL_CLASS" value="Tests\BitBag\SyliusCmsPlugin\Application\Kernel" />
         <server name="IS_DOCTRINE_ORM_SUPPORTED" value="true"/>
         <server name="EXPECTED_RESPONSE_DIR" value="../Responses/Expected"/>
-        <env name="APP_SECRET" value="''" />
-        <env name="APP_ENV" value="test"/>
```
Comment (marekrzytki) [inline phpunit.xml.dist:19]: Is this option not required in php unit 10.5?

File: tests/Application/Kernel.php [hunk 4]
```diff
@@ -53,74 +40,38 @@ public function registerBundles(): iterable
         }
     }
 
-    protected function configureContainer(ContainerBuilder $container, LoaderInterface $loader): void
+    private function configureContainer(ContainerBuilder $container, LoaderInterface $loader): void
```
Comment (marekrzytki) [inline tests/Application/Kernel.php:43]: Why private?

File: tests/Application/Kernel.php [hunk 5]
```diff
@@ -53,74 +40,38 @@ public function registerBundles(): iterable
...
     {
         $container->addResource(new FileResource($this->getProjectDir() . '/config/bundles.php'));
         $container->setParameter('container.dumper.inline_class_loader', true);
         $confDir = $this->getProjectDir() . '/config';
-        $syliusDir = $this->getProjectDir() . '/config/sylius/' . SyliusKernel::MAJOR_VERSION . '.' . SyliusKernel::MINOR_VERSION;
```
Comment (marekrzytki) [inline tests/Application/Kernel.php:61]: Why we are removing that? I'm pretty sure it's still required to load configuration files depending on Sylius version (1.12 or 1.13)

File: tests/Application/Kernel.php [hunk 6]
```diff
@@ -53,74 +40,38 @@ public function registerBundles(): iterable
...
 
         $loader->load($confDir . '/{packages}/*' . self::CONFIG_EXTS, 'glob');
         $loader->load($confDir . '/{packages}/' . $this->environment . '/**/*' . self::CONFIG_EXTS, 'glob');
         $loader->load($confDir . '/{services}' . self::CONFIG_EXTS, 'glob');
         $loader->load($confDir . '/{services}_' . $this->environment . self::CONFIG_EXTS, 'glob');
-        if (is_dir($syliusDir)) {
```
Comment (marekrzytki) [inline tests/Application/Kernel.php:67]: As mentioned above

File: tests/Application/Kernel.php [hunk 7]
```diff
@@ -53,74 +40,38 @@ public function registerBundles(): iterable
...
     {
-        if ($this->isTestEnvironment() && class_exists(MockerContainer::class)) {
+        if ($this->isTestEnvironment()) {
             return MockerContainer::class;
         }
 
         return parent::getContainerBaseClass();
     }
 
-    protected function getContainerLoader(ContainerInterface $container): DelegatingLoader
```
Comment (marekrzytki) [inline tests/Application/Kernel.php:96]: Why we are removing this method?

File: tests/Application/Kernel.php [hunk 8]
```diff
@@ -53,74 +40,38 @@ public function registerBundles(): iterable
...
-        foreach ($this->getConfigurationDirectories() as $confDir) {
-            $this->loadRoutesConfiguration($routes, $confDir);
-        }
-    }
-
-    private function loadRoutesConfiguration(RoutingConfigurator $routes, string $confDir): void
-    {
-        $routes->import($confDir . '/{routes}/*' . self::CONFIG_EXTS);
-        $routes->import($confDir . '/{routes}/' . $this->environment . '/**/*' . self::CONFIG_EXTS);
-        $routes->import($confDir . '/{routes}' . self::CONFIG_EXTS);
```
Comment (marekrzytki) [inline tests/Application/Kernel.php:84]: Why we are removing this methods?

File: tests/Application/Kernel.php [hunk 9]
```diff
@@ -53,74 +40,38 @@ public function registerBundles(): iterable
...
     }
 
     protected function getContainerBaseClass(): string
     {
-        if ($this->isTestEnvironment() && class_exists(MockerContainer::class)) {
+        if ($this->isTestEnvironment()) {
```
Comment (marekrzytki) [inline tests/Application/Kernel.php:57]: You are sure this class always exist?

File: tests/Application/config/bundles.php [hunk 10]
```diff
@@ -48,17 +50,16 @@
...
     SyliusLabs\DoctrineMigrationsExtraBundle\SyliusLabsDoctrineMigrationsExtraBundle::class => ['all' => true],
-    BitBag\SyliusCmsPlugin\BitBagSyliusCmsPlugin::class => ['all' => true],
-    Symfony\WebpackEncoreBundle\WebpackEncoreBundle::class => ['all' => true],
-    Sylius\Bundle\MailerBundle\SyliusMailerBundle::class => ['all' => true],
+    SyliusLabs\Polyfill\Symfony\Security\Bundle\SyliusLabsPolyfillSymfonySecurityBundle::class => ['all' => true],
     League\FlysystemBundle\FlysystemBundle::class => ['all' => true],
-    Sylius\Calendar\SyliusCalendarBundle::class => ['all' => true],
+    Symfony\WebpackEncoreBundle\WebpackEncoreBundle::class => ['all' => true],
+    FOS\CKEditorBundle\FOSCKEditorBundle::class => ['all' => true],
+    BitBag\SyliusCmsPlugin\BitBagSyliusCmsPlugin::class => ['all' => true],
```
Comment (marekrzytki) [inline tests/Application/config/bundles.php:64]: Why do we add plugin bundle there?

File: tests/Application/config/packages/liip_imagine.yaml [hunk 11]
```diff
@@ -1,10 +1,4 @@
 liip_imagine:
-    loaders:
```
Comment (marekrzytki) [inline tests/Application/config/packages/liip_imagine.yaml:2]: It's connected with Sylius update?

File: tests/Application/package.json [hunk 12]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (marekrzytki) [inline tests/Application/package.json:1]: Currently single package json in sufficient?

File: tests/Application/config/sylius/1.12/packages/security.yaml
Comment (marekrzytki) [inline tests/Application/config/sylius/1.12/packages/security.yaml:1]: Why we are removing this file? Also where is updated version of security for Sylius 1.13? https://github.com/Sylius/Sylius/blob/1.14/UPGRADE-1.13.md#:~:text=Due%20to%20changes%20in%20API%20paths%2C%20the%20security%20configuration%20has%20been%20changed%3A

File: tests/Application/config/sylius/1.12/packages/jms_serializer.yaml
Comment (marekrzytki) [inline tests/Application/config/sylius/1.12/packages/jms_serializer.yaml:1]: Why we are removing this config?

File: tests/Application/config/sylius/1.12/bundles.php
Comment (marekrzytki) [inline tests/Application/config/sylius/1.12/bundles.php:1]: Why we are removing this config?

**Review by marekrzytki — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (SzymonKostrubiec) [reply composer.json:12]: I have restored this dependency

File: phpunit.xml.dist (hunk 3 above)
Comment (SzymonKostrubiec) [reply phpunit.xml.dist:19]: Restored to original state

File: tests/Application/Kernel.php (hunk 4 above)
Comment (SzymonKostrubiec) [reply tests/Application/Kernel.php:43]: I have restored kernel file

File: tests/Application/config/bundles.php (hunk 10 above)
Comment (SzymonKostrubiec) [reply tests/Application/config/bundles.php:64]: It's required to use plugin correctly from test/Application,

File: tests/Application/package.json (hunk 12 above)
Comment (SzymonKostrubiec) [reply tests/Application/package.json:1]: Yes, the frontend library is sufficient to run the CmsPlugin

File: tests/Application/config/sylius/1.13/packages/security.yaml [hunk 13]
```diff
@@ -0,0 +1,124 @@
+security:
```
Comment (marekrzytki) [inline tests/Application/config/sylius/1.13/packages/security.yaml:1]: This file is not compatible with security changes from Sylius upgrade file: https://github.com/Sylius/Sylius/blob/1.14/UPGRADE-1.13.md -> point 11.
Also I think You are missing include in 1.13 bundles for Sylius\Abstraction\StateMachine\SyliusStateMachineAbstractionBundle::class => ['all' => true],

**Review by marekrzytki — CHANGES_REQUESTED** [review]

**Comment by Prometee** [comment]
@marekrzytki @SzymonKostrubiec thank you for not mentioning me in the Release note, like I already mentioned: you are not engaging comunity by replacing the original PR creator.

**Pull request merged**